0.8 Docs Changes (#2024)
hashicorp/boundary: website/content/docs/operations/health.mdx

---
layout: docs
page_title: Boundary Health Endpoints
description: |-
  Verify the Boundary controller server is up and able to receive requests
---

## Boundary Health Endpoints

Boundary provides health monitoring through the `/health` path using a listener

> with the `"ops"` purpose. By default a listener with that purpose runs on port

Reviewer: ```suggestion
with the `"ops"` purpose. By default, a listener with that purpose runs on port
```

`9203`. See the [example configuration](#example-configuration) section for an
example listener stanza in a `config.hcl` file.

### Requirements

> For the controller health endpoint to be enabled, any Boundary instance must be

Reviewer: ```suggestion
To enable the controller health endpoint, any Boundary instance must be
```

started with a controller. That is, a `controller` block and a `purpose = "api"`

> listener must be defined in Boundary's configuration file. Additionally, an

Reviewer: ```suggestion
listener must be defined in Boundary's configuration file. Additionally, a
```

`purpose = "ops"` listener must also be defined in Boundary's configuration

> file. Under these conditions the ops server (which hosts the controller health

Reviewer: ```suggestion
file. Under these conditions, the ops server (which hosts the controller health
```

api) will be exposed.

> ### Shutdown Grace Period

Reviewer: For the first sentence, my suggestion is to break it out into multiple sentences. Here's my suggestion, but please feel free to nix if my attempt to provide more clarity changed the intent of the sentence. Also, we'd want to avoid starting out a sentence with "Optionally."

**Suggestion**:
When the controller health endpoint is enabled, Boundary may be configured to change the controller health response to `503 Service Unavailable`. The controller health response will only change upon receiving a shutdown signal. An amount of time may be configured before the shutdown process begins.

> Optionally, when the controller health endpoint is enabled Boundary can be

Reviewer: I'd like to suggest a change with this sentence to break it up into multiple sentences so that it reads well.  Please feel free to nix my suggestion if it doesn't make sense or I've somehow changed the intent of the sentence. First off, typically it's best to avoid starting a sentence off with 'Optionally.' 

Here's my suggestion:
When the controller health endpoint is enabled, Boundary may be configured to change the controller health response to a`403 Server Unavailable`. The controller health response will only change upon receiving a shutdown signal. An amount of time may be configured before the shutdown process begins.

configured to change the controller health response to `503 Service Unavailable`
upon receiving a shutdown signal, and wait a configurable amount of time before
starting the shutdown process.
 
This feature is designed to integrate with load balancers to reduce the risk of
an outgoing Boundary instance causing disruption to incoming requests.
 
In this state, Boundary is still capable of processing requests as normal, but
will report as unhealthy through the controller health endpoint. In
load-balanced environments, this would cause this "unhealthy" instance to be

> removed from the pool of instances eligible to handle requests, reducing the

Reviewer: ```suggestion
removed from the pool of instances eligible to handle requests, and thereby, reducing the
```
